### PR TITLE
awscli: 1.18.150 -> 1.18.185

### PR DESCRIPTION
**Pull request by lovek323**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lovek323** [comment]
I'm not sure if I've done the different version of the `colorama` package correctly.

**Comment by SuperSandro2000** [comment]
> I'm not sure if I've done the different version of the `colorama` package correctly.

Is it possible to overwrite the version and hash of the original coloroma package or use an newer version without causing issues?

**Comment by lovek323** [comment]
> Is it possible to overwrite the version and hash of the original coloroma package or use an newer version without causing issues?

It seems that a quite a few other packages depend on it, and I'm moving _backwards_ by a patch number, so I thought it was probably wise to do it only for this specific package. Using a newer version won't fit the requirements for the `awscli` tool.

**Comment by SuperSandro2000** [comment]
That did not go as planned. Sorry about the pings.

**Comment by lukegb** [comment]
> It seems that a quite a few other packages depend on it, and I'm moving _backwards_ by a patch number, so I thought it was probably wise to do it only for this specific package. Using a newer version won't fit the requirements for the `awscli` tool.

I'd be tempted to patch the `setup.py` in `patchPhase` to be less strict about the version of colorama that's in use rather than packaging a separate point release. It's not at all clear that awscli actually needs specifically 0.4.3, and it's easier on us if we don't have multiple disparate versions of packages floating around.

**Comment by lovek323** [comment]
Thanks, @lukegb, I'll look into that.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -1273,6 +1273,8 @@ in {
 
   colorama = callPackage ../development/python-modules/colorama { };
 
+  colorama043 = callPackage ../development/python-modules/colorama/0.4.3.nix { };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix:1276]: please call this directly from awscli

File: pkgs/tools/admin/awscli/default.nix [hunk 2]
```diff
@@ -38,7 +38,7 @@ in with py.pkgs; buildPythonApplication rec {
     bcdoc
     s3transfer
     six
-    colorama
+    colorama043
```
Comment (FRidh) [inline pkgs/tools/admin/awscli/default.nix:41]: I find it highly unlikely that it is incompatible with a different patch release version. Please test patching out the pinning first.

**Review by FRidh — CHANGES_REQUESTED** [review]
1 commit per expressions/package modified

**Comment by SuperSandro2000** [comment]
@lovek323 Please fix the merge conflict.

File: pkgs/tools/admin/awscli/default.nix (hunk 2 above)
Comment (lovek323) [reply pkgs/tools/admin/awscli/default.nix:50]: Yep, will do, just waiting to have the time.

**Comment by jonringer** [comment]
also please address @FRidh comments

**Comment by lovek323** [comment]
Looks like this has already been fixed. Thanks for all your help, but I'm closing for now.

**Pull request closed without merging**